### PR TITLE
Fix build args integer issue

### PR DESCRIPTION
**Pull request by mtgoncalves1**
## Overview
Closes https://gitlab.com/architect-io/architect-cli/-/issues/645  
Fix issue with build args value that occurred when converting a docker-compose.yml to architect.yml.

## Changes
If build args is a map, cast each value to string.

## Tests
docker-compose.yml  
```
version: "3"
services:
  app:
    build:
      context: .
      args:
        INT_ARG: 1
        BOOL_ARG: true
```

Run `architect-local init --from-compose=docker-compose.yml test`.   
Should generate the following architect.yml 
```
name: test
services:
  app:
    build:
      args:
        INT_ARG: '1'
        BOOL_ARG: 'true'
      context: .
    reserved_name: app
```

### CODE REVIEW

File: src/common/docker-compose/converter.ts [hunk 1]
```diff
@@ -123,7 +123,9 @@ export class ComposeConverter {
           }
         }
       } else {
-        build.args = compose_build.args;
+        for (const [key, value] of Object.entries(compose_build.args as Dictionary<string>)) {
+          build.args[key] = value.toString();
```
Comment (mueschm) [inline src/common/docker-compose/converter.ts:127]: Is the `toString` needed if the Dictionary is already casting everything to a string?

**Review by mueschm — APPROVED** [review]
Looks good and makes sense to me

File: src/common/docker-compose/converter.ts [hunk 2]
```diff
@@ -123,7 +123,9 @@ export class ComposeConverter {
           }
         }
       } else {
```
Comment (ryan-cahill) [inline src/common/docker-compose/converter.ts:125]: Since this block is using specific logic and not just setting what's passed in to the build args anymore, we should do something like 

```ts
else if (compose_build.args instanceof Object && Object.keys(compose_build.args).length)
```

in order to make sure that the new code will actually be passed a useful dictionary

File: src/common/docker-compose/converter.ts (hunk 1 above)
Comment (ryan-cahill) [reply src/common/docker-compose/converter.ts:127]: We shouldn't use `Dictionary<string>` here because what's actually being passed in is a `Dictionary<any>`. The `as` is just telling TS that everything is okay and it'll only see string values, but this isn't actually casting the values. This compose file 

```yml
version: 3
services:
  test-service:
    build:
      args:
        API_VERSION: $API_VERSION
        EMPTY:
```

will cause this to explode because of what I mentioned above. We'll need a little more logic here to handle all of the cases that a user could throw at us, for example:
```ts
build.args[key] = !!value ? value.toString() : value;
```

We should include a test for an empty arg here too

**Review by ryan-cahill — CHANGES_REQUESTED** [review]

File: src/common/docker-compose/converter.ts (hunk 1 above)
Comment (mtgoncalves1) [reply src/common/docker-compose/converter.ts:127]: Sounds good. I'll fix it.

**Review by ryan-cahill — APPROVED** [review]
